test/integration/targets/ec2_elb/tasks/main.yml

```diff
@@ -0,0 +1,536 @@
+        security_token: "{{ security_token }}"
+        state: present
+        cross_az_load_balancing: true
+        security_group_ids:
+          - "{{ setup_sg.group_id }}"
+        subnets:
+          - "{{ setup_subnet.subnet.id }}"
+        listeners:
+          - protocol: http
+            load_balancer_port: 80
+            instance_port: 80
+          - protocol: http
+            load_balancer_port: 8080
+            instance_port: 8080
+        health_check:
+            ping_protocol: tcp
+            #ping_protocol: http
+            ping_port: 80
+            ping_path: "/"
+            response_timeout: 5
+            interval: 30
+            unhealthy_threshold: 4
+            healthy_threshold: 2
+
+    # ============================================================
+
+    # SETUP COMPLETE
+
+    # ============================================================
+
+    - name: run module without providing required options
+      ec2_elb:
+        region: "{{ aws_region }}"
+        aws_access_key: "{{ aws_access_key }}"
+        aws_secret_key: "{{ aws_secret_key }}"
+        security_token: "{{ security_token }}"
+      register: result
+      ignore_errors: yes
+
+    - name: assert task would have failed
+      assert:
+        that:
+          - "not result.changed"
+          - "result.msg.startswith('missing required arguments:')"
+
+    # ============================================================
+
+    - name: test credential parameters
+      ec2_elb:
+        ec2_elbs: "{{ lb_name }}"
+        instance_id: i-xxxxxxxx
+        region: "{{ aws_region }}"
+        state: present
+      register: result
+      ignore_errors: true
+
+    - name: assert credential parameters
+      assert:
+        that:
+           - "result.failed"
+           - "result.msg == 'Unable to locate credentials'"
+
+    # ============================================================
+
+    - name: test invalid region parameter
+      ec2_elb:
+        ec2_elbs: "{{ lb_name }}"
+        instance_id: i-xxxxxxxx
+        region: "asdf querty 1234"
+        aws_access_key: "{{ aws_access_key }}"
+        aws_secret_key: "{{ aws_secret_key }}"
+        security_token: "{{ security_token }}"
+        state: present
+      register: result
+      ignore_errors: true
+
+    - name: assert invalid region parameter
+      assert:
+        that:
+           - "result.failed"
+
+    # ============================================================
+
+    - name: test partial credentials
+      ec2_elb:
+        region: "{{ aws_region }}"
+        aws_access_key: "{{ aws_access_key }}"
+        # aws_secret_key: "{{ aws_secret_key }}"
+        # security_token: "{{ security_token }}"
+        state: present
+        ec2_elbs: "{{ lb_name }}"
+        instance_id: "{{ setup_instance.instance_ids[0] }}"
+      register: result
+      ignore_errors: true
+
+    - name: assert task failed
+      assert:
+        that:
+          - "result.failed"
```

Comment:
The preferred way to check failure is to use the filter, so there's no error if that attribute is undefined. Just `result | failed`